language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/workspace/BallerinaWorkspaceManager.java

```diff
@@ -1009,7 +1009,10 @@ private ProjectPair createProject(Path filePath, String operationName) {
         try {
             Project project;
             BuildOptions options = BuildOptions.builder()
-                    .setOffline(CommonUtil.COMPILE_OFFLINE).setSticky(true).build();
+                    .setOffline(CommonUtil.COMPILE_OFFLINE)
+                    .setSticky(true)
+                    .targetDir(String.valueOf(Files.createTempDirectory("ballerina-cache" + System.nanoTime())))
```

Comment:
Let's use the `System.getProperty("java.io.tmpdir")` to construct the target. This will eliminate the need to handle the IOException. We can do this as follows:
- Introduce `ProjectUtils.getTemporaryTargetPath`
- The above method will construct a path using the system property and return
e.g:
```
public static Path getTemporaryTargetPath() {
      return Paths.get(System.getProperty("java.io.tmpdir")).resolve("ballerina-cache" + System.nanoTime()");
}
```

Reply:
Introduced a new method in `ProjectUtils` and used it here.